src/index.js

```diff
@@ -216,21 +240,36 @@ const compare = (condition, value, attributes) => {
 const reduceRule = (rule, attributes) => {
   const result = {};
 
-  for (const [name, condition] of Object.entries(rule)) {
-    const values = getAttributeValues(attributes, name.split('.'));
+  for (let [pathToCheck, condition] of Object.entries(rule)) {
+    const { pathToCheck: newPathToCheck, condition: newCondition } =
+      maybeReverseCondition(pathToCheck, condition, attributes);
+    pathToCheck = newPathToCheck;
+    condition = newCondition;
+
+    // When we already know the value of the target, we replace it with an
+    // in-line value.
+    if (condition.target) {
+      const inLineTargetValue = getAttribute(attributes, condition.target);
+
+      if (inLineTargetValue) {
+        condition.value = inLineTargetValue;
+        delete condition.target;
```

Comment:
Is it definitely okay to mutate here? Is it possible we'd be mutating any inputs to the top-level APIs like `enforce` or `reduce(...)`?

Reply:
It's ok to mutate here because `reduceRule` is only called within `reduceRules`, which is already immutable. I wonder if we should take the `cloneDeep`s one level down so `reduceRule` is the one that's always immutable, no matter where it's used.

Reply:
+1, that seems sane

Reply:
I think I prefer to leave the immutability at the `reduceRules` level since we're only really using `reduceRule` there and that way we don't clone `attributes` unnecessarily a bunch of times.

Reply:
@swain see my last thought here.